TicTacToeServer/TicTacToeServer.Core/Program.cs

```diff
@@ -1,15 +1,166 @@
+            var directory = ParseArg(DirectoryOption, args);
+            var log = ParseArg(LogOption, args);
+            var arrayClean = IsArrayClean(args);
+            try
+            {
+                if (log != null && arrayClean)
+                    io.Log = log;
+                if (port != null && directory != null && arrayClean)
+                    return MakedirectoryServer(port, directory, io);
+                if (port != null && arrayClean)
+                    return HelloWorldServer(port, io);
+                io.Print(WrongNumberOfArgs());
+                return null;
+            }
+            catch (Exception)
+            {
+                io.Print("Another Server is running on that port");
+                return null;
+            }
+        }
+
+        public static IMainServer MakedirectoryServer(string chosenPort, string homeDirectory, IPrinter io)
+        {
+            var cleanHomeDir = homeDirectory.Replace('\\', '/');
+            var port = PortWithinRange(chosenPort, io);
+            if (port == -1) return null;
+            if (!VaildDrive(cleanHomeDir, io)) return null;
+            var endPoint = new IPEndPoint((IPAddress.Loopback), port);
+            var zSocket = new ZSocket(endPoint);
+            var properties = new ServerProperties(cleanHomeDir, 
+                new DirectoryProcessor(), new FileProcessor(), port,
+                new HttpResponse(), new ServerTime(), io,
+                new TicTacToeGame(new User(), new Ai(),MakeSettings()));
+            return new MainServer(zSocket, properties, 
+                new HttpServiceFactory(new Service404()),
+                new List<string>() { "TicTacToeServer.Core" },
+                new List<Assembly>() { Assembly.GetExecutingAssembly() });
+        }
+
+        public static IMainServer HelloWorldServer(string port, IPrinter io)
+        {
+            var portConverted = PortWithinRange(port, io);
+            if (portConverted == -1) return null;
+            var endPoint = new IPEndPoint((IPAddress.Loopback), portConverted);
+            var zSocket = new ZSocket(endPoint);
+            var properties = new ServerProperties(null, new DirectoryProcessor(), new FileProcessor(), portConverted,
+                new HttpResponse(), new ServerTime(), io,
+                new TicTacToeGame(new User(), new Ai(), MakeSettings()));
+            return new MainServer(zSocket, properties, 
+                new HttpServiceFactory(new Service404()),
+                new List<string>() { "TicTacToeServer.Core" },
+                new List<Assembly>() { Assembly.GetExecutingAssembly() });
+        }
+
+        private static bool VaildDrive(string dir, IPrinter io)
+        {
+            if (Directory.Exists(dir))
+            {
+                return true;
+            }
+            io.Print("Not a vaild directory");
+            return false;
+        }
+
+        private static int PortWithinRange(string port, IPrinter io)
```

Comment:
Method does not make sense.  Am I return a port or checking if it is within range.